src/docbuilder/options.rs

```diff
@@ -94,16 +93,13 @@ impl DocBuilderOptions {
 
     pub fn check_paths(&self) -> Result<()> {
         if !self.destination.exists() {
-            return Err(err_msg("Destination path not exists"));
+            bail!("destionation path '{}' does not exist", self.destination.display());
         }
         if !self.chroot_path.exists() {
-            return Err(err_msg("Chroot path not exists"));
+            bail!("chroot path '{}' does not exist", self.chroot_path.display());
         }
         if !self.crates_io_index_path.exists() {
-            return Err(err_msg("crates.io-index path not exists"));
-        }
-        if !self.crates_io_index_path.exists() {
```

Comment:
Can you replace this with a check for `self.sources_path` instead? Looks like that path is not being checked.